src/data/setoid/partition.lean

```diff
@@ -191,15 +191,16 @@ instance partition.partial_order : partial_order (subtype (@is_partition α)) :=
 
 variables (α)
 
-/-- The order-preserving bijection between equivalence relations and partitions of sets. -/
+/-- The order-preserving bijection between equivalence relations on a type `α`, and
+  partitions of `α` into subsets. -/
 protected def partition.order_iso :
-  setoid α ≃o subtype (@is_partition α) :=
+  setoid α ≃o {C : set (set α) // is_partition C} :=
```

Comment:
This is less delta-reduced which is _maybe_ worse. If that is a concern, then this would still be more readable than the original, while being not only defeq but syntactically eq:
```suggestion
  setoid α ≃o subtype (is_partition : set (set α) → Prop) :=
```

Reply:
I'll let another reviewer decide.

Reply:
I'm of course happy if this doesn't get merged, but basically this is a result which is never used elsewhere in mathlib and I'm simply suggesting that it's rewritten to be more readable, for teaching purposes. Basically I only changed it because I was writing lectures on equivalence relations.

Reply:
@PatrickMassot @adamtopaz This file is used in LTE, right? Would this change break anything there?

Reply:
If it breaks anything then it will be easily fixable.

Reply:
I'm in favor of merging this.